src/main/js/Widgets/AJAXDialog.js

```diff
@@ -343,11 +352,32 @@ class AJAXDialog {
         .go();
   }
 
+  _handleClick(event) {
+    if (event.target === Overlay.instance.overlay.domElement) {
+      if (this._isTopDialog() && this.dismissible) {
+        Utils.stopEvent(event);
+        this.close();
+      }
+    }
+  }
+
+  _handleKeyDown(event) {
+    if (this._isTopDialog()) {
+      if (event.key === 'Escape' && this.dismissible) {
+        Utils.stopEvent(event);
+        this.close();
+      }
+    }
+  }
+
   _initializeDialog() {
     this.element.query(this.options.closeButtonElementSelector).each(function(e) {
       e.addEventListener('click', this._handleCloseClickEvent);
     }.bind(this));
 
+    document.addEventListener('keydown', this._handleKeyDown);
```

Comment:
Good point.  Updated.  Left the logic in the handler just in case someone adds other handling.  Or in the unlikely case someone calls withDismissible(false) after setHTML is called.